### PR TITLE
Signal Handling fixes for controller

### LINKED ISSUE #118: Appsody run doesn't properly terminate the process
**Describe the bug**
It seems that stopping `appsody run` (and others) doesn't properly terminate the underlying process. I haven't looked into what happened under the hood just yet, however the output of, for example, the Maven process in `java-microprofile` doesn't print the usual stopping that it does when hitting `Ctrl + C` in the command line, and neither does `appsody run` wait for the embedded server to stop.

If the process (e.g. the running server) is just being killed without having received a `TERM` signal then resources and connections as not properly closed and released.

**To Reproduce**
1. Run `appsody run` in a `java-microprofile` project and hit `Ctrl + C`
2. The process just stops and doesn't seem to shutdown the server.
3. Compare with running the `mvn` command locally and hit `Ctrl + C`

**Expected behavior**
Stopping whatever is running with a proper `TERM` signal. This might require proper handling of Unix signals to potential child processes.

**If applicable please specify:**
 - CLI version: `appsody 0.2.5`
 - Stack you are using: `java-microprofile` and home-grown stacks

**Additional context**
https://blog.phusion.nl/2015/01/20/docker-and-the-pid-1-zombie-reaping-problem/

### PR DESCRIPTION
**Pull request by kewegner**
Fixes #118

### CODE REVIEW

**Comment by kewegner** [comment]
Do not merge this PR unless https://github.com/appsody/controller/pull/51 has been merged first.
Controller version will need to be updated for the release.

File: cmd/dev_common.go [hunk 1]
```diff
@@ -292,8 +289,12 @@ func commonCmd(config *devCommonConfig, mode string) error {
 			// TODO presumably you can query the error itself
 			error := fmt.Sprintf("%s", err)
 			//Linux and Windows return a different error on Ctrl-C
-			if error == "signal: interrupt" || error == "exit status 2" {
-				Info.log("Closing down, development environment was interrupted.")
+			if error == "signal: interrupt" || error == "signal: terminated" || error == "exit status 2" {
+				Info.log("Closing down, development environment was interrupted will now sleep 10 secondsx.")
```
Comment (chilanti) [inline cmd/dev_common.go]: Fix the typo.

File: cmd/dev_common.go [hunk 2]
```diff
@@ -290,7 +290,7 @@ func commonCmd(config *devCommonConfig, mode string) error {
 			error := fmt.Sprintf("%s", err)
 			//Linux and Windows return a different error on Ctrl-C
 			if error == "signal: interrupt" || error == "signal: terminated" || error == "exit status 2" {
-				Info.log("Closing down, development environment was interrupted will now sleep 10 secondsx.")
+				Info.log("Closing down, development environment was interupted will now sleep 10 secondsx.")
```
Comment (chilanti) [inline cmd/dev_common.go]: Now there are two typos: `interupted` and `secondsx`.

Comment (kewegner) [reply cmd/dev_common.go]: fixed

File: cmd/dev_common.go [hunk 3]
```diff
@@ -291,9 +288,14 @@ func commonCmd(config *devCommonConfig, mode string) error {
 			// 'signal: interrupt'
 			// TODO presumably you can query the error itself
 			error := fmt.Sprintf("%s", err)
+			Debug.log("CLI exit error is:  ", error)
 			//Linux and Windows return a different error on Ctrl-C
-			if error == "signal: interrupt" || error == "exit status 2" {
-				Info.log("Closing down, development environment was interrupted.")
+			if error == "signal: interrupt" || error == "signal: terminated" || error == "exit status 2" {
+				Info.log("Closing down, development environment was interrupted will now sleep 10 seconds.")
```
Comment (kylegc) [inline cmd/dev_common.go]: I don't like saying `will now sleep 10 seconds`.
Lets just leave it at
`Closing down, development environment was interrupted.`

File: cmd/run.go [hunk 4]
```diff
@@ -20,7 +20,9 @@ import (
 
 func dockerStop(imageName string, dryrun bool) error {
 	cmdName := "docker"
-	cmdArgs := []string{"stop", imageName}
+	signalInterval := "10" // numnrt of seconds to wait prior to sending SIGKILL
+	Info.logf("Performing docker stop. Shutdown will occur in %s seconds", signalInterval)
```
Comment (kylegc) [inline cmd/run.go]: Let's remove this message. The execAndWait will already log the command `docker stop -t 10` so the user can figure out what that means.

File: cmd/dev_common.go (hunk 3 above)
Comment (kewegner) [reply cmd/dev_common.go]: changed

File: cmd/run.go (hunk 4 above)
Comment (kewegner) [reply cmd/run.go]: changed.

File: cmd/dev_common.go (hunk 2 above)
Comment (kewegner) [reply cmd/dev_common.go]: fixed.

**Pull request merged**